crates/ty_python_semantic/resources/mdtest/type_qualifiers/initvar.md

```diff
@@ -0,0 +1,124 @@
+# `dataclasses.InitVar`
+
+From the Python documentation on [`dataclasses.InitVar`]:
+
+If a field is an `InitVar`, it is considered a pseudo-field called an init-only field. As it is not
+a true field, it is not returned by the module-level `fields()` function. Init-only fields are added
+as parameters to the generated `__init__()` method, and are passed to the optional `__post_init__()`
```

Comment:
Ideally we would also verify that the signature of `__post_init__` is correct: that it includes all dataclass fields _and_ all `InitVar` pseudo-fields in the parameters list (https://github.com/python/mypy/issues/15498). Doesn't need to be done in this PR, of course!

Reply:
> I don't think there's really any practical use for having an `InitVar` field if you _don't_ have a `__post_init__` method, right? So we could possibly even consider having a `WARN`-severity lint that complains if you do that.

I think you could also just have a custom `__init__` instead? In that case, the `InitVar` declaration would only be documentation, I guess.

Reply:
> It would not just be documentation -- the field would not be included in the generated `__repr__` of instances, it would not be taken into account by the generated `__eq__` method of the class, and (if it's a slotted dataclass) it won't be added to the dataclass's `__slots__`.

Right, but you could also just not mention the init-only field at all in that case, right? :-)